lib/includes/wslay/wslay.h

```diff
@@ -145,13 +145,18 @@ enum wslay_opcode {
 #define wslay_is_ctrl_frame(opcode) ((opcode >> 3) & 1)
 
 /*
- * Macros that returns reserved bits: RSV1, RSV2, RSV3.  These macros
- * assumes that rsv is constructed by ((RSV1 << 2) | (RSV2 << 1) |
- * RSV3)
+ * Macros that represent and return reserved bits: RSV1, RSV2, RSV3.
+ * These macros assumes that rsv is constructed by ((RSV1 << 2) |
+ * (RSV2 << 1) | RSV3)
  */
-#define wslay_get_rsv1(rsv) ((rsv >> 2) & 1)
-#define wslay_get_rsv2(rsv) ((rsv >> 1) & 1)
-#define wslay_get_rsv3(rsv) (rsv & 1)
+#define WSLAY_RSV_NONE ((uint8_t) 0)
+#define WSLAY_RSV1_BIT (((uint8_t) 1) << 2)
+#define WSLAY_RSV2_BIT (((uint8_t) 1) << 1)
+#define WSLAY_RSV3_BIT (((uint8_t) 1) << 0)
+
+#define wslay_get_rsv1(rsv) (rsv & WSLAY_RSV1_BIT)
+#define wslay_get_rsv2(rsv) (rsv & WSLAY_RSV2_BIT)
+#define wslay_get_rsv3(rsv) (rsv & WSLAY_RSV3_BIT)
```

Comment:
Technically, this changes the return value of this macro.  It would be safe to retain original return value here since we have no way to know how the exiting application uses these macros.